dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,371 @@
+import {
+  Box,
+  InputAdornment,
+  LinearProgress,
+  makeStyles,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TextField,
+  TextFieldProps,
+  Tooltip,
+} from "@material-ui/core";
+import { SearchOutlined } from "@material-ui/icons";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import Pagination from "@material-ui/lab/Pagination";
+import dayjs from "dayjs";
+import React, { useEffect, useState } from "react";
+import { getEvents } from "../service/event";
+import { Event } from "../type/event";
+import { useFilter } from "../util/hook";
+import { StatusChip } from "./StatusChip";
+
+export enum SeverityLevel {
+  INFO = "INFO",
+  DEBUG = "DEBUG",
+  WARNING = "WARNING",
+  ERROR = "ERROR",
+  TRACING = "TRACING",
+} // to maintain and sync with event.proto
+export enum SourceType {
+  COMMON = "COMMON",
+  CORE_WORKER = "CORE_WORKER",
+  GCS = "GCS",
+  RAYLET = "RAYLET",
+  CLUSTER_LIFECYCLE = "CLUSTER_LIFECYCLE",
+  AUTOSCALER = "AUTOSCALER",
+  JOBS = "JOBS",
+  SERVE = "SERVE",
+} // to maintain and sync with event.proto
```

Comment:
let's put this in the `event.d.ts` file?

Reply:
Sounds good